fix: medusa-service generated update functions always returns an array even if single object is passed.

dwene opened the pull request:
Related to the change with Medusa Internal Service here: https://github.com/medusajs/medusa/pull/12715/files#diff-4d287d33cf799750dfc3957d3c22d87a5d369ca15bbbc7db29265fc864cedfadL248

Context: all of my medusa-service generated update functions are always returning an array, forcing me to have to check the response myself.

ex:


```
export const updateThemeStep = createStep(
  'update-theme-step',
  async (input: UpdateCheckoutThemeWorkflowStepInput, { container }) => {
    const themeModuleService: ThemeModuleService = container.resolve(CHECKOUT_THEME_MODULE);

    const { channel_id, ...updateThemeData } = input;

    const originalTheme = await queryCheckoutTheme({ container, themeId: input.id });

    const theme = await themeModuleService.updateCheckoutThemes(updateThemeData);

    if (Array.isArray(theme)) return new StepResponse(theme[0], originalTheme);

    return new StepResponse(theme, originalTheme);
  },
  async (originalTheme, { container }) => {
    if (!originalTheme) return;

    const themeModuleService: ThemeModuleService = container.resolve(CHECKOUT_THEME_MODULE);

    await themeModuleService.updateCheckoutThemes(originalTheme);
  }
);
```

dwene committed:
fix: medusa-service generated update always returns an array even if object is passed

dwene committed:
add: changeset

dwene committed:
chore: cleanup

olivermrbl commented on packages/core/utils/src/modules-sdk/medusa-service.ts:
```diff
@@ -243,7 +243,7 @@ export function MedusaService<
           sharedContext: Context = {}
         ): Promise<T | T[]> {
-          const serviceData = Array.isArray(data) ? data : [data]
+          const serviceData = data
```
@adrien2p, should we add a few tests for this before we merge?

adrien2p replied:
Yea definitely 💯 forgot to mention it. Also, removing the service data const as it does not do anything anymore

olivermrbl replied:
cc @dwene

dwene committed:
Merge branch 'develop' into fix/medusa-service-update-always-returns-array

codegen-sh[bot] committed:
Add comprehensive tests for medusa-service create/update methods

- Add tests for single object input/output consistency
- Add tests for array input/output consistency
- Add proper mocks for create/update service methods
- Fix update method to preserve input type (single object vs array)
- Tests validate the bug fix where update method was always returning arrays

dwene committed:
chore: cleanup

dwene committed:
Merge branch 'fix/medusa-service-update-always-returns-array' of github.com:lambda-curry/medusa into fix/medusa-service-update-always-returns-array

dwene committed:
Merge branch 'develop' into fix/medusa-service-update-always-returns-array

dwene committed:
fix: create service impl

dwene commented:
@olivermrbl OK all the tests are passing now. Looks like create had the same issue.

dwene committed:
Merge branch 'develop' into fix/medusa-service-update-always-returns-array

dwene commented:
Looks like this was fixed by a core team member in PR https://github.com/medusajs/medusa/pull/13225 so I'm going to close this.

Thanks for fixing this @carlos-r-l-rodrigues !
